crates/block-producer/src/test_utils/block.rs

```diff
@@ -71,30 +68,31 @@ pub async fn build_expected_block_header(
     )
 }
 
-/// Builds the "actual" block header; i.e. the block header built using the Miden VM, used in the
-/// node
-pub async fn build_actual_block_header(
-    store: &MockStoreSuccess,
-    batches: Vec<ProvenBatch>,
-) -> BlockHeader {
-    let updated_accounts: Vec<_> =
-        batches.iter().flat_map(|batch| batch.account_updates().iter()).collect();
-    let produced_nullifiers: Vec<Nullifier> =
-        batches.iter().flat_map(ProvenBatch::produced_nullifiers).collect();
-
-    let block_inputs_from_store: BlockInputs = store
-        .get_block_inputs(
-            updated_accounts.iter().map(|(&account_id, _)| account_id),
-            produced_nullifiers.iter(),
-            iter::empty(),
-        )
-        .await
-        .unwrap();
-
-    let (block_witness, _) = BlockWitness::new(block_inputs_from_store, &batches).unwrap();
-
-    BlockProver::new().prove(block_witness).unwrap()
-}
+// Note: Commented as it is unused atm. Is it worth fixing it?
```

Comment:
If its unused we should nuke it. We can always resurrect it.

Probably a remnant of when I yanked the block building tests with the mempool rewrite.

I assume this should be better done using `MockChain` in the future